src/Tizen.Applications.ComponentBased/Tizen.Applications.ComponentBased.Common/ComponentFactoryBase.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Tizen.Applications.CoreBackend;
+using static Interop.CBApplication;
+using static Tizen.Applications.CoreBackend.DefaultCoreBackend;
+
+namespace Tizen.Applications.ComponentBased.Common
+{
+    internal abstract class ComponentFactoryBase
+    {
+        public IList<BaseComponent> ComponentInstances { get; }
+        public Type ComponentClassType { get; }
+        public ComponentType ComponentType { get; }
+        public string ComponentId { get; }
+        public ComponentBasedApplication Parent { get; }
```

Comment:
It seem to circular dependency, 

BaseComponentFactoryBase -> ComponentBaseApplication
 :  BaseComponentFactoryBase.Parent 

ComponentBaseApplication -> BaseComponentFactoryBase 
 :  ComponentBasedApplication._componentFactories

Reply:
I changed it into protected.